libfwupdplugin/fu-bluez-device.c

```diff
@@ -350,18 +409,140 @@ fu_bluez_device_read (FuBluezDevice *self, const gchar *uuid, GError **error)
  * Since: 1.5.7
  **/
 gchar *
-fu_bluez_device_read_string (FuBluezDevice *self, const gchar *uuid, GError **error)
+fu_bluez_device_read_string (FuBluezDevice *self, const gchar *uuid_str, GError **error)
 {
-	GByteArray *buf = fu_bluez_device_read (self, uuid, error);
+	GByteArray *buf = fu_bluez_device_read (self, uuid_str, error);
 	if (buf == NULL)
 		return NULL;
 	return (gchar *) g_byte_array_free (buf, FALSE);
 }
 
+/**
+ * fu_bluez_device_connect_signal:
+ * @self: A #FuBluezDevice
+ * @uuid_str: The UUID, e.g. `00cde35c-7062-11eb-9439-0242ac130002`
+ * @signal_name: The signal to connect, e.g. `g-properties-changed`
+ * @callback: The handler for this signal
+ * @error: A #GError, or %NULL
+ *
+ * Connects to a signal of a DBus object identified by a UUID and
+ * registers a callback to handle it.
+ * If successful, it saves the signal id in the uuid object.
+ *
+ * Returns: %TRUE if the signal connection was successful.
+ *
+ * Since: 1.5.7
+ **/
+gboolean
+fu_bluez_device_connect_signal (FuBluezDevice *self,
+				const gchar *uuid_str,
+				const gchar *signal_name,
+				gpointer callback,
+				GError **error)
+{
+	FuBluezUuid *uuid;
+	GDBusProxy *proxy;
+
+	uuid = fu_bluez_device_get_uuid (self, uuid_str, error);
+	if (uuid == NULL)
+		return FALSE;
+
+	proxy = fu_bluez_device_get_uuid_proxy (self, uuid_str, error);
+	if (proxy == NULL)
+		return FALSE;
+
+	uuid->signal_id = g_signal_connect (proxy,
+					    signal_name,
+					    G_CALLBACK (callback), NULL);
+	if (uuid->signal_id <= 0) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_NOT_SUPPORTED,
+			     "Can't connect to signal %s of UUID %s", signal_name, uuid_str);
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+/**
+ * fu_bluez_device_start_notify:
```

Comment:
do we need this? couldn't we call it automatically when calling `fu_bluez_device_connect_signal`?

Reply:
Yes, I agree we need _start() and _stop() -- I just think we should _start() by default.